edgelet/build/linux/package.sh

```diff
@@ -0,0 +1,499 @@
+
+            'arm32v7')
+                DOCKER_IMAGE='edgebuilds.azurecr.io/gcc-linaro-7.3.1-2018.05-x86_64_arm-linux-gnueabihf:centos_7.5-1'
+                ;;
+
+            'aarch64')
+                DOCKER_IMAGE='edgebuilds.azurecr.io/gcc-linaro-7.3.1-2018.05-x86_64_aarch64-linux-gnu:centos_7.5-1'
+                ;;
+        esac
+
+        CMAKE_ARGS="$CMAKE_ARGS '-DOPENSSL_DEPENDS_SPEC=openssl-libs'"
+        ;;
+
+    'debian8')
+        case "$PACKAGE_ARCH" in
+            'amd64')
+                DOCKER_IMAGE='debian:8-slim'
+                ;;
+
+            # Debian 8 doesn't have cross-compiler packages in its main repo
+            #
+            # The emdebian repos have cross-compiler packages but they cannot be installed because of broken dependencies.
+            # >gcc-4.9-arm-linux-gnueabihf : Depends: libgcc-4.9-dev:armhf (= 4.9.2-10) but 4.9.2-10+deb8u2 is to be installed
+            #
+            # emdebian is also not maintained any more, not in the least because Debian 9+ have cross-compiler packages in the main repo.
+            #
+            # So stick with the linaro compiler for now.
+            'arm32v7')
+                DOCKER_IMAGE='edgebuilds.azurecr.io/gcc-linaro-7.3.1-2018.05-x86_64_arm-linux-gnueabihf:debian_8.11-1'
+                ;;
+
+            'aarch64')
+                # Like the comment in packages.yaml says, Debian 8 aarch64 is not LTS, and doesn't have any official or ports repos.
+                #
+                # So don't build it at all.
+                ;;
+        esac
+
+        CMAKE_ARGS="$CMAKE_ARGS '-DOPENSSL_DEPENDS_SPEC=libssl1.0.0'"
+        ;;
+
+    'debian9')
+        DOCKER_IMAGE='debian:9-slim'
+        CMAKE_ARGS="$CMAKE_ARGS '-DOPENSSL_DEPENDS_SPEC=libssl1.1'"
+        ;;
+
+    'debian10')
+        DOCKER_IMAGE='debian:10-slim'
+        CMAKE_ARGS="$CMAKE_ARGS '-DOPENSSL_DEPENDS_SPEC=libssl1.1'"
+        ;;
+
+    'ubuntu16.04')
+        DOCKER_IMAGE='ubuntu:16.04'
+        CMAKE_ARGS="$CMAKE_ARGS '-DOPENSSL_DEPENDS_SPEC=libssl1.0.0'"
+        ;;
+
+    'ubuntu18.04')
+        DOCKER_IMAGE='ubuntu:18.04'
+        CMAKE_ARGS="$CMAKE_ARGS '-DOPENSSL_DEPENDS_SPEC=libssl1.1'"
+        ;;
+esac
+
+if [ -z "$DOCKER_IMAGE" ]; then
+    echo "Unrecognized target [$PACKAGE_OS.$PACKAGE_ARCH]" >&2
+    exit 1
+fi
+
+case "$PACKAGE_ARCH" in
+    'amd64')
+        MAKE_FLAGS="DPKGFLAGS='-b -us -uc -i'"
+        ;;
+
+    'arm32v7')
+        CMAKE_ARGS="$CMAKE_ARGS -DCPACK_DEBIAN_PACKAGE_ARCHITECTURE=armhf"
+        CMAKE_ARGS="$CMAKE_ARGS -DCPACK_RPM_PACKAGE_ARCHITECTURE=armv7hl"
+
+        RUST_TARGET='armv7-unknown-linux-gnueabihf'
+        ;;
+
+    'aarch64')
+        CMAKE_ARGS="$CMAKE_ARGS -DCPACK_DEBIAN_PACKAGE_ARCHITECTURE=arm64"
+        CMAKE_ARGS="$CMAKE_ARGS -DCPACK_RPM_PACKAGE_ARCHITECTURE=aarch64"
+
+        RUST_TARGET='aarch64-unknown-linux-gnu'
+        ;;
+esac
+
+if [ -n "$RUST_TARGET" ]; then
+    RUST_TARGET_COMMAND="rustup target add $RUST_TARGET &&"
```

Comment:
>rustup target add [](start = 25, length = 17)

no need to install rust-src?  or was that needed for cross?

Reply:
It's not even needed for cross unless you need to compile for an arch that needs xargo, which is none of the arches we used it for.